File: longest-lines/longest.py
import sys

with open(sys.argv[1]) as file:
  lines = file.readlines()
n = int(lines[0])
lines = filter(lambda x: len(x) > 0, lines[1:])

print('\n'.join(sorted(lines, key=lambda x: len(x), reverse=True)[0:n]))

File: longest-lines/longest.rb
module LongestLines
  def self.run(filename = ARGV[0])
    File.open(filename) do |file|
      num = file.readline.to_i
      lines = []

      file.each_line do |line|
        line.strip!
        next if line.empty?

        lines.push line
      end

      lines.sort! { |a, b| b.length <=> a.length }
      lines = lines[0...num]
      lines.each { |line| puts line }
    end
  end
end

LongestLines.run if __FILE__ == $0